### PR TITLE
RD-5756 Migrate `shared` components to TypeScript - part 1 

### PR DESCRIPTION
**Pull request by qooban**
## Description

First part of `shared` components migration to TypeScript.

In another PR(s) the following components are planned to be migrated:
* `app/components/shared/widgets/PageContent.tsx` (requires #2292 )
* `app/components/shared/widgets/Widget.tsx` (more complex change, need to fix widget definition types hierarchy)
* `app/components/shared/MaintenanceModeModal.tsx` (bigger change, I'm planning refactor in maintenance-related properties in Redux store first)

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
Added component test for `Graph` component.

## Documentation
N/A

### CODE REVIEW

File: app/components/layout/GridItem.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ interface GridItemProps {
     y?: number;
     width?: number;
     height?: number;
+    maximized?: boolean;
```
Comment (qooban) [inline app/components/layout/GridItem.tsx:14]: Required for `app/components/shared/widgets/WidgetsGrid.tsx`.

File: app/components/shared/graphs/Graph.tsx [hunk 2]
```diff
@@ -61,89 +111,10 @@ import {
  *
  * ## Usage
  *
- * ### Bar chart
```
Comment (qooban) [inline app/components/shared/graphs/Graph.tsx:64]: Examples converted to tests - `test/cypress/components/app/Graph_spec.tsx`.

**Review by kubama — APPROVED** [review]
+1 for adding component test :)

**Comment by qooban** [comment]
> +1 for adding component test :)

Yeah, I think we should do that more often 🙂

**Pull request merged**